### PR TITLE
Fix unwraps in `send` mod

### PR DESCRIPTION
**Pull request by DanGould**
Unnecessary unwraps were left in the send mod when the big v2 changes were merged. This addresses those.

~This is urgent to release 0.13.0~

### CODE REVIEW

File: payjoin/src/send/mod.rs [hunk 1]
```diff
@@ -486,10 +494,12 @@ impl Serialize for RequestContext {
...
         state.serialize_field("psbt", &self.psbt.to_string())?;
         state.serialize_field("endpoint", &self.endpoint.as_str())?;
-        let ohttp_string = self
-            .ohttp_config
-            .as_ref()
-            .map_or("".to_string(), |config| bitcoin::base64::encode(config.encode().unwrap()));
+        let ohttp_string = self.ohttp_config.as_ref().map_or(Ok("".to_string()), |config| {
+            config
+                .encode()
+                .map_err(|e| serde::ser::Error::custom(format!("ohttp-keys encoding error: {}", e)))
```
Comment (jbesraa) [inline payjoin/src/send/mod.rs:500]: whats the motivation for using `Error::custom` here and not doing ` .map_err(|e| InternalCreateRequestError::V2(e.into()))?` as above?

**Review by jbesraa — COMMENTED** [review]
Added a small question.
Also I think this one was missed https://github.com/payjoin/rust-payjoin/pull/161/files#diff-e1a2a0997d5241c211b358d2ea2be04fc1a4aa680ac543480188d7b4ab63ff72L832

File: payjoin/src/send/mod.rs (hunk 1 above)
Comment (DanGould) [reply payjoin/src/send/mod.rs:500]: We have to return `Result<Serializer::Ok, Serializer::Error>` here where CreateRequestError is not Serializer::Error. Above is in a function that returns `Result<(Request, ContextV2), CreateRequestError>` where that would be InternalCreateRequestError has a conversion into CreateRequestError.

**Comment by DanGould** [comment]
re: 

> Also I think this one was missed https://github.com/payjoin/rust-payjoin/pull/161/files#diff-e1a2a0997d5241c211b358d2ea2be04fc1a4aa680ac543480188d7b4ab63ff72L832

If you look at the blame that expect is from the initial commit of the repo and checked in `RequestBuilder::build()`: https://github.com/payjoin/rust-payjoin/pull/161/files#diff-e1a2a0997d5241c211b358d2ea2be04fc1a4aa680ac543480188d7b4ab63ff72R336

**Review by jbesraa — APPROVED** [review]

**Pull request merged**